### ISSUE TITLE
Unable to compile moveit_task_constructor

### ISSUE THREAD

**Issue by hcostelha** [ros-planning/moveit_task_constructor#377]
Hi,

I was trying to follow the MoveIt getting started with ROS2 Humble in Ubuntu 22.04, using the binary install. Due to the problem show [here](https://answers.ros.org/question/404556/issue-with-binary-install-of-moveit2-with-ros2-humble-in-ubuntu-2202/), I had to install ros-humble-pilz-industrial-motion-planner_2.5.3-1jammy.20220801.185125_amd64.deb from [here](http://packages.ros.org/ros2-testing/ubuntu/pool/main/r/ros-humble-pilz-industrial-motion-planner/).

I then created a workspace and cloned the moveit2_tutorial,  moveit_task_constructor and rosparam_shortcuts (packages not available in the Ubuntu ROS2 repositories) using, respectively
`git clone https://github.com/ros-planning/moveit2_tutorials -b humble`
`git clone https://github.com/ros-planning/moveit_task_constructor -b ros2`
`vcs import < moveit_task_constructor/.repos`

Next I did
`source /opt/ros/humble/setup.bash`
and compiled it using
`colcon build --mixin release`
but I got the following errors:
```
~/ws_moveit2$ colcon build --mixin release
Starting >>> moveit_task_constructor_msgs
Starting >>> rviz_marker_tools
Starting >>> rosparam_shortcuts
Finished <<< rviz_marker_tools [0.21s]
Finished <<< rosparam_shortcuts [0.22s]
Finished <<< moveit_task_constructor_msgs [0.64s]
Starting >>> moveit_task_constructor_core
--- stderr: moveit_task_constructor_core
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp: In constructor ‘moveit::task_constructor::solvers::CartesianPath::CartesianPath()’:
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:62:47: error: ‘IKCostFn’ is not a member of ‘kinematics::KinematicsBase’
   62 |         p.declare<kinematics::KinematicsBase::IKCostFn>("kinematics_cost_fn", kinematics::KinematicsBase::IKCostFn(),
      |                                               ^~~~~~~~
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:62:107: error: ‘IKCostFn’ is not a member of ‘kinematics::KinematicsBase’
   62 |   p.declare<kinematics::KinematicsBase::IKCostFn>("kinematics_cost_fn", kinematics::KinematicsBase::IKCostFn(),
      |                                                                                                     ^~~~~~~~

/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp: In member function ‘virtual bool moveit::task_constructor::solvers::CartesianPath::plan(const PlanningSceneConstPtr&, const moveit::core::LinkModel&, const Isometry3d&, const moveit::core::JointModelGroup*, double, robot_trajectory::RobotTrajectoryPtr&, const Constraints&)’:
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:106:51: error: ‘IKCostFn’ is not a member of ‘kinematics::KinematicsBase’
  106 |             props.get<kinematics::KinematicsBase::IKCostFn>("kinematics_cost_fn"));
      |                                                   ^~~~~~~~
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:106:60: error: no matching function for call to ‘moveit::task_constructor::PropertyMap::get<<expression error> >(const char [19]) const’
  106 |             props.get<kinematics::KinematicsBase::IKCostFn>("kinematics_cost_fn"));
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In file included from /home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/solvers/planner_interface.h:43,
                 from /home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/solvers/cartesian_path.h:41,
                 from /home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:39:
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/properties.h:324:18: note: candidate: ‘template<class T> const T& moveit::task_constructor::PropertyMap::get(const string&) const’
  324 |         const T& get(const std::string& name) const {
      |                  ^~~
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/properties.h:324:18: note:   template argument deduction/substitution failed:
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:106:60: error: template argument 1 is invalid
  106 |             props.get<kinematics::KinematicsBase::IKCostFn>("kinematics_cost_fn"));
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In file included from /home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/solvers/planner_interface.h:43,
                 from /home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/solvers/cartesian_path.h:41,
                 from /home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:39:
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/properties.h:332:18: note: candidate: ‘template<class T> const T& moveit::task_constructor::PropertyMap::get(const string&, const T&) const’
  332 |         const T& get(const std::string& name, const T& fallback) const {
      |                  ^~~
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/include/moveit/task_constructor/properties.h:332:18: note:   template argument deduction/substitution failed:
/home/hcostelha/ws_moveit2/src/moveit_task_constructor/core/src/solvers/cartesian_path.cpp:106:60: error: template argument 1 is invalid
  106 |             props.get<kinematics::KinematicsBase::IKCostFn>("kinematics_cost_fn"));
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/moveit_task_constructor_core.dir/build.make:230: src/CMakeFiles/moveit_task_constructor_core.dir/solvers/cartesian_path.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:254: src/CMakeFiles/moveit_task_constructor_core.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< moveit_task_constructor_core [7.07s, exited with code 2]

Summary: 3 packages finished [7.90s]
  1 package failed: moveit_task_constructor_core
  1 package had stderr output: moveit_task_constructor_core
  4 packages not processed
```

I saw that the latest change to movei_task_constructor, as shown [here](https://github.com/ros-planning/moveit_task_constructor/commit/1e65027b26744dccd02bd7d7ac4d0856916cfde5), affects this part of the code that is giving me an error. Should I update something else? Do I have to update everything to version 2.5.3? Any other thoughts?

I also tried compiling everything from source, but the compiler keeps crashing (probably due to limited memory available), evenn when using just 1 worker, so that does not seem to be an alternative for me.

Thanks and best regards

**Comment by hcostelha** [comment]
I just saw that a change was applied to moveit_ore that went to version 2.5.3, so I probably need to update everything to 2.5.3 to get a coherent installation. I will do that and report back.

**Comment by hcostelha** [comment]
So, I have updated all (and only) the moveit packages to the version available in _testing_, which is on version 2.5.3. After doing so, the moveit_task_constructor compiled successfully. No I have another compilation problem in the tutorials, but is another story.

Sorry for the entropy. Best regards.
